charts/tidb-cluster/templates/config/_grafana-config.tpl

```diff
@@ -204,13 +204,13 @@ root_url = %(protocol)s://%(domain)s:%(http_port)s/
 #################################### Anonymous Auth ##########################
 [auth.anonymous]
 # enable anonymous access
-;enabled = false
+enabled = {{ .Values.monitor.grafana.anonymousEnabled | default true }}
```

Comment:
I really want us to avoid the busywork of eventually templating every variable here. I would prefer to back out grafana changes from this PR and separately change them as per #449. Or you can take the changes in #449 here. With #449 we wouldn't change things here and the values would have `"auth.anonymous.enabled": true`, etc.

Reply:
OK, I'll remove the configmap.

Reply:
I think its still okay to load the grafana.ini file if someone wants to configure grafana through that. Its just that we won't template the file from helm.

Reply:
What if both grafana.ini and environment variables are set? Leaving two different configurations confuses users.

Reply:
Grafana has already chosen to allow multiple ways to configure their application. Our goal should be to just transparently pass-through configuration to it. Grafana documents how different configuration options interact.

Reply:
But the configmap is hidden in the template directory, users can hardly configure it. So it's unnecessary to leave it there.

Reply:
We have several configurations there including that for pd/tikv/tidb so it is important that we document for users how to modify these files. Alternatively we could document how they can add a config file.

I found the helm chart for grafana to be confusing because one generates it from helm values very similar to the env variables: ` --set 'grafana\.ini.auth\.anonymous.enabled'='true' `